api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go

```diff
@@ -654,8 +658,19 @@ func (i *K8sTargetProvider) upsertDeployment(ctx context.Context, namespace stri
 	}
 
 	waitErr := i.pollCheck(ctx, false, func() error {
-		_, err := i.Client.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
-		return err
+		d, err := i.Client.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
```

Comment:
Glad to see we added polling for deployment. I think we need to check the observedGeneration == generation at first. Example in helm: https://github.com/helm/helm/blob/3a3e3846ca9c929a6966583b461181e70f19bc13/pkg/kube/ready.go#L109
Besides, how about the polling for service type, like what helm tool does - https://github.com/helm/helm/blob/3a3e3846ca9c929a6966583b461181e70f19bc13/pkg/kube/ready.go#L253.

Reply:
fixed. follow the same check pattern with helm